src/matcha_ml/services/analytics_service.py

```diff
@@ -0,0 +1,81 @@
+"""The analytics service interface.
+
+This approach to collecting usage data was inspired by ZenML; source: https://github.com/zenml-io/zenml/blob/main/src/zenml/utils/analytics_utils.py
+"""
+import functools
+from enum import Enum
+from time import time
+from typing import Any, Callable, Optional
+
+from segment import analytics
+
+from matcha_ml.services.global_parameters_service import GlobalParameters
+from matcha_ml.services.matcha_state import MatchaStateService
+
+analytics.write_key = "qwBKAvY6MEUvv5XIs4rE07ohf5neT3sx"
+
+
+class AnalyticsEvent(str, Enum):
+    """Analytics event enum class."""
+
+    PROVISION = "provision"
+    DESTROY = "destroy"
+    GET = "get"
+
+
+def track(event_name: AnalyticsEvent) -> Callable[..., Any]:
+    """Track decorator for tracking user analytics with Segment.
+
+    Args:
+        event_name (str): Name of the command or event being called, e.g. 'provision'
+    """
+
+    def decorator(func: Callable[..., Any]) -> Callable[..., Any]:
+        """Decorator function.
+
+        Args:
+            func (Callable[..., Any]): The function that is being decorated
+        """
+
+        @functools.wraps(func)
+        def inner(*args: Any, **kwargs: Any) -> Any:
+            """Inner decorator function."""
+            global_params = GlobalParameters()
+
+            if not global_params.analytics_opt_out:
+                ts = time()
```

Comment:
Nit: I would suggest using `time.perf_counter` instead of `time`.

Reference: https://stackoverflow.com/a/66037261 and https://discuss.pytorch.org/t/why-time-time-in-python-is-inaccurte/94274